workflow.go

```diff
@@ -0,0 +1,165 @@
+package workflow
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net/http"
+	"net/url"
+	"time"
+)
+
+// Workflow defines the trigger functions
+type Workflow struct {
+	api         string
+	accessToken string
+	eventType   string
+	client      *http.Client
+}
+
+// Params collects repository credentials and workflow information
+type Params struct {
+	BaseURL     string
+	Repo        string
+	Owner       string
+	AccessToken string
+	EventType   string
+	MaxRetries  int
+	Delay       time.Duration
+}
+
+// Configure creates an instance of Workflow
+func Configure(p Params) *Workflow {
+	host := "api.github.com"
+	if p.BaseURL != "" {
+		host = p.BaseURL
+	}
+
+	u := url.URL{
+		Scheme: "https",
+		Host:   host,
+		Path:   fmt.Sprintf("/repos/%s/%s/dispatches", p.Owner, p.Repo),
+	}
+
+	maxRetries := 1
+	if p.MaxRetries > 1 {
+		maxRetries = p.MaxRetries
+	}
+
+	delay := 5 * time.Second
+	if p.Delay > 0*time.Second {
```

Comment:
I think you can just use `0` here? It's an [untyped constant](https://blog.golang.org/constants) so it should be comparable to a time.Duration.
```suggestion
	if p.Delay > 0 {
```

Reply:
Also you may want to make this a higher minimum, if a user specifices `time.Nanosecond`, you'll have a very short duration. Maybe the minimum should be 1 second?